jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/CreateBranchTest.groovy

```diff
@@ -0,0 +1,158 @@
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.jtalks.jcommune.test
+
+import org.jtalks.common.model.entity.Component
+import org.jtalks.common.model.permissions.GeneralPermission
+import org.jtalks.jcommune.test.model.Branch
+import org.jtalks.jcommune.test.model.User
+import org.jtalks.jcommune.test.service.BranchService
+import org.jtalks.jcommune.test.service.ComponentService
+import org.jtalks.jcommune.test.service.GroupsService
+import org.jtalks.jcommune.test.service.UserService
+import org.jtalks.jcommune.test.utils.Branches
+import org.jtalks.jcommune.test.utils.exceptions.ProcessingException
+import org.jtalks.jcommune.test.utils.exceptions.ValidationException
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.test.context.ContextConfiguration
+import org.springframework.test.context.transaction.TransactionConfiguration
+import org.springframework.test.context.web.WebAppConfiguration
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.setup.MockMvcBuilders
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.context.WebApplicationContext
+import spock.lang.Specification
+
+import javax.annotation.Resource
+import javax.servlet.Filter
+
+import static org.apache.commons.lang.RandomStringUtils.randomAlphabetic
+
+/**
+ * @author Mikhail Stryzhonak
+ */
+@WebAppConfiguration
+@ContextConfiguration(locations = [
+        'classpath:/org/jtalks/jcommune/model/entity/applicationContext-dao.xml',
+        'classpath:/org/jtalks/jcommune/model/entity/applicationContext-properties.xml',
+        'classpath:/org/jtalks/jcommune/service/applicationContext-service.xml',
+        'classpath:/org/jtalks/jcommune/service/security-service-context.xml',
+        'classpath:/org/jtalks/jcommune/service/email-context.xml',
+        'classpath:/org/jtalks/jcommune/web/applicationContext-controller.xml',
+        'classpath:security-context.xml',
+        'classpath:spring-dispatcher-servlet.xml',
+        'classpath:/org/jtalks/jcommune/web/view/test-configuration.xml'
+])
+@TransactionConfiguration(transactionManager = 'transactionManager', defaultRollback = true)
+@Transactional
+class CreateBranchTest extends Specification {
+
+    @Autowired
+    private WebApplicationContext ctx
+    @Resource(name = 'testFilters')
+    List<Filter> filters
+
+    @Autowired
+    private Branches branches;
+    @Autowired
+    private ComponentService componentService
+    @Autowired
+    private UserService userService
+    @Autowired
+    private BranchService branchService
+    @Autowired
+    private GroupsService groupsService
+
+    private MockMvc mockMvc
+
+    private Component forum
+
+    def setup() {
+        mockMvc = MockMvcBuilders.webAppContextSetup(ctx)
```

Comment:
Let's crete this mockMvc in Spring context and set it into `branches` there (in Spring).